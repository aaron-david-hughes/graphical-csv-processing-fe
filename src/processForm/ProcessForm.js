import React from 'react';
import Input from "../generalPurposeComponents/input/Input";

class ProcessForm extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            title: DefaultFilename
        }
    }

    setTitle(title) {
        if (title.length <= 0) {
            title = DefaultFilename;
        }

        if (!title.endsWith('.zip')) {
            title = title + '.zip';
        }

        this.setState({
            title
        });
    }

    render() {
        return <form
            id='processForm'
            style={{
                width: '100%'
            }}
            onSubmit={e => this.props.onSubmitForm(e, this.state.title)}
        >
            <div
                style={{
                    width: '100%'
                }}
            >
                <div
                    style={{
                        width: '100%',
                        display: 'inline-flex'
                    }}
                >
                    <Input
                        title='Filename'
                        style={{width: '100%'}}
                        input={
                            <input
                                id='filename'
                                type='text'
                                onChange={e => this.setTitle(e.target.value)}
                            />
                        }
                    />
                </div>
                <div style={{margin: '5px'}}>
                    <p className='ow'>
                        <i className='fa fa-info-circle'/>
                        {` Filename to be used: ${this.state.title}`}
                    </p>
                </div>
                <button>Process</button>
            </div>
        </form>
    }
}

const DefaultFilename = 'Result.zip';

export default ProcessForm;