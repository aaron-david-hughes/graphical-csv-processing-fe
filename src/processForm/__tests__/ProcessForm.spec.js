import React from 'react';
import {shallow, mount} from 'enzyme';
import ProcessForm from "../ProcessForm";

describe('ProcessForm tests', () => {
    let props = {
        loading: jest.fn(),
        onSubmitForm: jest.fn(),
        isLoading: false,
        isGraphValid: () => true
    };

    let render;
    let instance;

    beforeEach(() => {
        render = shallow(<ProcessForm {...props} />);
        instance = render.instance();
    });

    describe('render tests', () => {
        it('should match snapshot when not processing', () => {
            expect(render).toMatchSnapshot();
        })

        it('should match snapshot when processing', () => {
            let localProps = {
                ...props,
                isLoading: true
            };

            render = shallow(<ProcessForm {...localProps} />);

            expect(render).toMatchSnapshot();
        });

        it('should call setTitle onChange of title input', () => {
            render = mount(<ProcessForm {...props} />);
            instance = render.instance();

            let setTitleSpy = jest.spyOn(instance, 'setTitle');

            render.find('#filename').simulate('change');

            expect(setTitleSpy).toHaveBeenCalled();
        });

        it('should call onSubmitForm on form submission', async () => {
            render = mount(<ProcessForm {...props} />);
            instance = render.instance();

            await render.find('#processForm').simulate('submit');

            expect(props.loading).toHaveBeenCalled();
            expect(props.onSubmitForm).toHaveBeenCalled();
        });
    });

    describe('instance tests', () => {
        describe('setTitle tests', () => {
            it('should set default title if title empty', () => {
                instance.setTitle('');
                expect(instance.state.title).toEqual('Result.zip');
            });

            it('should set suffix .zip onto end of title if not present', () => {
                instance.setTitle('test-result');
                expect(instance.state.title).toEqual('test-result.zip');
            });

            it('should set not suffix .zip onto end of title if already present', () => {
                instance.setTitle('test-result.zip');
                expect(instance.state.title).toEqual('test-result.zip');
            });
        });
    });
});