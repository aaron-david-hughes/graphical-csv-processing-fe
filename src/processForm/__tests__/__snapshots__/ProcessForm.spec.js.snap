// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProcessForm tests render tests should match snapshot when not processing 1`] = `
<form
  id="processForm"
  onSubmit={[Function]}
  style={
    Object {
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "display": "inline-flex",
          "width": "100%",
        }
      }
    >
      <Input
        input={
          <input
            id="filename"
            onChange={[Function]}
            type="text"
          />
        }
        style={
          Object {
            "width": "100%",
          }
        }
        title="Filename"
      />
    </div>
    <div
      style={
        Object {
          "margin": "5px",
        }
      }
    >
      <p
        className="ow"
      >
        <i
          className="fa fa-info-circle"
        />
         Filename to be used: Result.zip
      </p>
    </div>
    <button
      disabled={false}
      id="processButton"
      title="process"
    >
      process
    </button>
  </div>
</form>
`;

exports[`ProcessForm tests render tests should match snapshot when processing 1`] = `
<form
  id="processForm"
  onSubmit={[Function]}
  style={
    Object {
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "display": "inline-flex",
          "width": "100%",
        }
      }
    >
      <Input
        input={
          <input
            id="filename"
            onChange={[Function]}
            type="text"
          />
        }
        style={
          Object {
            "width": "100%",
          }
        }
        title="Filename"
      />
    </div>
    <div
      style={
        Object {
          "margin": "5px",
        }
      }
    >
      <p
        className="ow"
      >
        <i
          className="fa fa-info-circle"
        />
         Filename to be used: Result.zip
      </p>
    </div>
    <Loader
      color="#000000"
      css=""
      loading={true}
      size={30}
      speedMultiplier={1}
    />
  </div>
</form>
`;
